src/sqlfluff/core/templaters/dbt.py

```diff
@@ -248,9 +249,20 @@ def _unsafe_process(self, fname, in_str=None, config=None):
                 "by running `dbt compile` directly."
             )
 
-        raw_sliced, sliced_file, templated_sql = self.slice_file(
-            node.raw_sql, compiled_sql, config=config
-        )
+        exclude_rules = self.sqlfluff_config.__dict__["_configs"]["core"][
+            "exclude_rules"
+        ]
+
+        if re.search("L009", str(exclude_rules)):
+            raw_sliced, sliced_file, templated_sql = self.slice_file(
```

Comment:
I'd prefer to avoid duplicating this line of code. How about we add a line to compute a value for`trailing_newline` based on `exclude_rules`, then pass that value to `self.slice_file()`.